src/hotspot/share/code/debugInfo.cpp

```diff
@@ -242,7 +242,10 @@ void ObjectValue::print_fields_on(outputStream* st) const {
 // of the object.
 ObjectValue* ObjectMergeValue::select(frame& fr, RegisterMap& reg_map) {
   StackValue* sv_selector = StackValue::create_stack_value(&fr, &reg_map, _selector);
-  jint selector = sv_selector->get_int();
+  // On big endian platforms the jint is in the high part of the StackValue
+  intptr_t val = sv_selector->get_int();
+  jint selector = (jint)*((jint*)&val);
```

Comment:
You don't need outer cast but you may need `()` around `(&val)`. See the example:
https://github.com/openjdk/jdk/blob/master/src/hotspot/share/runtime/stackValueCollection.cpp#L29